lighthouse-core/audits/third-party-facades.js

```diff
@@ -0,0 +1,209 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/**
+ * @fileoverview Audit which identifies third-party code on the page which can be lazy loaded.
+ * The audit will recommend a facade alternative which is used to imitate the third party resource until it is needed.
+ *
+ * Entity: Set of domains which are used by a company or product area to deliver third party resources
+ * Product: Specific piece of software belonging to an entity. Entities can have multiple products.
+ * Facade: Placeholder for a product which looks likes the actual product and replaces itself with that product when the user needs it.
+ */
+
+/** @typedef {import("third-party-web").IEntity} ThirdPartyEntity */
+/** @typedef {import("third-party-web").IProduct} ThirdPartyProduct*/
+/** @typedef {import("third-party-web").IFacade} ThirdPartyFacade*/
+
+/** @typedef {{product: ThirdPartyProduct, entity: ThirdPartyEntity}} FacadableProduct */
+
+const Audit = require('./audit.js');
+const i18n = require('../lib/i18n/i18n.js');
+const thirdPartyWeb = require('../lib/third-party-web.js');
+const NetworkRecords = require('../computed/network-records.js');
+const MainResource = require('../computed/main-resource.js');
+const MainThreadTasks = require('../computed/main-thread-tasks.js');
+const ThirdPartySummary = require('./third-party-summary.js');
+
+const UIStrings = {
+  /** Title of a diagnostic audit that provides details about the third-party code on a web page that can be lazy loaded with a facade alternative. This descriptive title is shown to users when no resources have facade alternatives available. Lazy loading means loading resources is deferred until they are needed. */
```

Comment:
I think "third party" will need to be explained in the translation comment.

Reply:
starting to wonder if we should add a glossary to these comments

```
/**
 * @glossary
 * @word {Facade} Something blah blah blah
 * @word {Product} Something blah blah blah
 * @word {Third-Party} Something blah blah blah
 */
```

And then concat all of these definitions to every translator comment in this file when we collect strings.